docs/src/main/mdoc/datatypes/statet.md

```diff
@@ -0,0 +1,285 @@
+      else Right(Reservations(reservations :+ reservation))
+  }
+
+  final case class TableAlreadyReserved(
+      reservation: Reservation
+  ) extends RuntimeException(
+        s"${reservation.name} cannot be added because table number ${reservation.id.tableNumber} is already reserved for the ${reservation.id.hour}"
+      )
+
+  val emptyReservationSystem: Reservations = Reservations(List.empty)
+
+  def insertBooking(
+      reservation: Reservation
+  ): StateT[ThrowableOr, Reservations, Unit] =
+    StateT.modifyF[ThrowableOr, Reservations](_.insert(reservation))
+
+  def processBookings(
+      bookings: NonEmptyList[Reservation]
+  ): ThrowableOr[Reservations] =
+    bookings
+      .map(insertBooking)
+      .reduceLeft[StateT[ThrowableOr, Reservations, Unit]] {
+        case (acc, bookNext) => acc.productL(bookNext)
+      }
+      .runS(emptyReservationSystem)
+}
+```
+
+That's it, we can finally test the code above providing a simple
+example input and print out the result.
+
+```scala mdoc
+val bookings = NonEmptyList.of(
+  TableReservationSystem.Reservation(
+    TableReservationSystem
+      .ReservationId(tableNumber = 1, hour = LocalTime.parse("10:00:00")),
+    name = "Cristiano Ronaldo"
+  ),
+  TableReservationSystem.Reservation(
+    TableReservationSystem
+      .ReservationId(tableNumber = 2, hour = LocalTime.parse("10:00:00")),
+    name = "Lebron James"
+  ),
+  TableReservationSystem.Reservation(
+    TableReservationSystem
+      .ReservationId(tableNumber = 1, hour = LocalTime.parse("12:00:00")),
+    name = "Tiger Woods"
+  ),
+  TableReservationSystem.Reservation(
+    TableReservationSystem
+      .ReservationId(tableNumber = 2, hour = LocalTime.parse("12:00:00")),
+    name = "Roger Federer"
+  ),
+  TableReservationSystem.Reservation(
+    TableReservationSystem
+      .ReservationId(tableNumber = 3, hour = LocalTime.parse("13:00:00")),
+    name = "Elizabeth Alexandra Mary"
+  ),
+  TableReservationSystem.Reservation(
+    TableReservationSystem
+      .ReservationId(tableNumber = 1, hour = LocalTime.parse("16:00:00")),
+    name = "Chuck Norris"
+  ),
+  TableReservationSystem.Reservation(
+    TableReservationSystem
+      .ReservationId(tableNumber = 2, hour = LocalTime.parse("16:00:00")),
+    name = "Kobe Bryant"
+  ),
+  TableReservationSystem.Reservation(
+    TableReservationSystem
+      .ReservationId(tableNumber = 2, hour = LocalTime.parse("18:00:00")),
+    name = "Steven Seagal"
+  )
+)
+
+TableReservationSystem.processBookings(bookings)
+
+TableReservationSystem.processBookings(
+  bookings :+ TableReservationSystem.Reservation(
+    TableReservationSystem
+      .ReservationId(tableNumber = 1, hour = LocalTime.parse("16:00:00")),
+    name = "Bruce Lee"
+  )
+)
+```
+
+The full source code of this example can be found at this
+[gist](https://gist.github.com/benkio/baa4fe1d50751cd602c4175f1bb39f4d)
+or [scastie](https://scastie.scala-lang.org/MFHZpU9bRg2e73MLQY1dNQ)
+
+## Example: Hangman Game
```

Comment:
> Unfortunately, the rest of the example can't be shown here due to the

I think it's a bit strange to have 1.5 example :) maybe you can host this one in a personal website/gist, and we can link to from the cats site.

Reply:
There's a `gist` and a `scastie` at the end of the example as a "continuation". So you are suggesting to remove the "possible" explanation and just leave those links at the end of the documentation hinting something like:
"for a more advanced example check..."

Reply:
Thanks, yes, exactly. Imo it seems odd to start an example we can actually finish :)